### PR TITLE
mount: Fixes for both 9pfs and virtio-blk/scsi

### PR DESCRIPTION
**Pull request by sameo**
* Set `absSource` properly.
* Create 9pfs target directories in the guest.

### CODE REVIEW

**Comment by jodh-intel** [comment]
This doesn't compile for me?

**Comment by sameo** [comment]
@jodh-intel Sorry I was rushing things out...Fixed now.

File: mount.go [hunk 1]
```diff
@@ -57,14 +57,25 @@ var flagList = map[string]int{
 	"runbindable": unix.MS_UNBINDABLE | unix.MS_REC,
 }
 
+func createDestinationDir(dest string) error {
+	targetPath, _ := filepath.Split(dest)
+	if err := os.MkdirAll(targetPath, mountPerm); err != nil {
```
Comment (jodh-intel) [inline mount.go]: You could simplify this to:

```
return os.MkdirAll(targetPath, mountPerm)
```

**Comment by bergwolf** [comment]
LGTM

**Comment by bergwolf** [comment]
travis service is broken. Merge regardlessly.

**Comment by laijs** [comment]
It uses https://golang.org/pkg/path/filepath/#Split

if the destination is '/path/to/9pfs/dir', only the '/path/to/9pfs/' is created, not '/path/to/9pfs/dir'.
Is it deliberated?

@sameo

**Comment by sameo** [comment]
@laijs No, it's not deliberated. It works for us because we're always passing a `/` terminated path to our gRPC `Storage` messages. But we should make that code more robust by calling `Split` only when the mount point is not a directory.

**Comment by laijs** [comment]
for 9pfs case, I think it's better always to consider the destination is dir.

**Pull request merged**